### ISSUE TITLE
Problems with attachments on HTML email (error -10000)

### ISSUE THREAD

**Issue by abroess** [nevik/apple-mail-evernote-export#1]
Hi Patrick,
there is a problem with html email attachments. The script cannon count the attachments and even export them for appending to the Ervernote note. Any suggestions?

Alex

**Comment by lehnerpat** [comment]
Well! I'm quite surprise someone else has already found this script, hehe.

I've experienced more issues as well. What error are you getting?

If it's an error `-10000`, that is an issue in Mail and its interface to AppleScript. From what I've been able to find out, the mailbox index sometimes gets corrupted in a particular way, so that mails can still be handled fine in Mail, but interacting with attachments via AppleScript breaks, indicated but that error code.

The only fix for that, as far as I know, is to have Mail re-index your mailboxes. To do this, close Mail and delete any files starting with `Envelope Index` (usually three of them) in `~/Library/Mail/V2/MailData`. (Just to be safe, keep backups at first.) When you start Mail after this, it shows the "Import" dialog but does not require any interaction. Once your mails have been re-imported (it really just re-indexes them), the script should work fine where it broke before. I've experienced that some Smart Mailboxes don't properly work immediately after re-importing mails. Restarting Mail again once the import is done fixes this problem as well.

If you're experiencing other problems, let me know!
